### PR TITLE
Show ThemeToggler on supported themes only

### PR DESCRIPTION
**Pull request by MikeAlhayek**
The Theme toggler (Dark/Auto/Light) should be explicitly added to themes that support it. By default it should not be added to every frontend theme.

### CODE REVIEW

File: src/OrchardCore.Modules/OrchardCore.Themes/Drivers/ToggleThemeNavbarDisplayDriver.cs
Comment (Piedone) [inline src/OrchardCore.Modules/OrchardCore.Themes/Drivers/ToggleThemeNavbarDisplayDriver.cs:1]: Shouldn't this rather be in `TheAdmin`? Also, there are three `ToggleTheme.cshtml`s, and two scripts this doesn't seem right: Either the feature is reasonable reusable, and then we shouldn't have slightly different copies of the driver and template, or it's not, but then there shouldn't be anything the Themes module.

There's the toggler JS that seems reusable (and indeed reused) at least.

**Review by Piedone — CHANGES_REQUESTED** [review]
Was this causing any issues? Since nothing was displayed e.g. under TheBlog theme. Because if not, it would be easier to just permanently display it in `Detail`, as it was (but not `DetailAdmin`), rather than reimplementing the driver in every theme where we need it.

**Comment by MikeAlhayek** [comment]
It is showing up on every theme that uses the NavBar which is wrong because not every theme supports theme toggle. So we should not by default expose it since the default implementation is for Bootstrap 5.3 only

If someone designs a bs5.3 theme and wants to support theme toggle, they should add the toggler using a driver. But we should not just display it always. 

The issue that I ran into is that I am creating a theme that does not need toggler, but the toggler is showing up for me. Not something we should do

**Comment by Piedone** [comment]
OK, I see.

File: src/OrchardCore.Modules/OrchardCore.Themes/Drivers/ToggleThemeNavbarDisplayDriver.cs
Comment (MikeAlhayek) [reply src/OrchardCore.Modules/OrchardCore.Themes/Drivers/ToggleThemeNavbarDisplayDriver.cs:1]: @Piedone 
When I developed this, I made it reusable for any BS 5.3 theme. However, not all BS 5.3-based themes require a toggler, even though BS 5.3 itself supports themes. The `ToggleThemeNavbarDisplayDriver` driver can be relocated to `TheAdmin`, and we can also move `ToggleTheme.DetailAdmin.cshtml`. However, the `ToggleTheme.cshtml` file should remain in the Themes directory to maintain its reusability. This ensures that users who are using BS 5.3 and want the default toggle implementation can still access it without issues along with `theme-manager` script which is needed for the toggler to work.

Comment (Piedone) [reply src/OrchardCore.Modules/OrchardCore.Themes/Drivers/ToggleThemeNavbarDisplayDriver.cs:1]: OK!

**Comment by MikeAlhayek** [comment]
@Piedone anything else to add here or should we merge it?

**Comment by Piedone** [comment]
See https://discord.com/channels/551136772243980291/1246954405014671513/1317165890181140560 I'll review soon.

**Review by Piedone — APPROVED** [review]

**Pull request merged**